File: javascript/hamming/hamming.js
var Hamming = function() {
}

Hamming.prototype.compute = function(firstSequence, secondSequence) {

    var numDifferences = 0;
    if (firstSequence.length != secondSequence.length) {
        throw new Error('DNA strands must be of equal length.');
    }
    for (var i = 0; i < firstSequence.length; i++){
      if (firstSequence[i] != secondSequence[i]){
        numDifferences++;
      }
    }
    return numDifferences;
}

module.exports = Hamming;

File: python/hamming/hamming.py
def distance(firstSequence, secondSequence):

    numDifferences = 0
    if len(firstSequence) != len(secondSequence):
        raise ValueError('DNA strands must be of equal length.')
    for i in range (0,len(firstSequence)):
      if firstSequence[i] != secondSequence[i]:
        numDifferences += 1
    return numDifferences;
    pass